client/blocks/term-form-dialog/index.jsx

```diff
@@ -145,6 +146,13 @@ class TermFormDialog extends Component {
 
 	init( props ) {
 		if ( ! props.term ) {
+			if ( props.searchTerm && props.searchTerm.trim().length ) {
+				this.setState( assign( {}, this.constructor.initialState, {
```

Comment:
We could combine them into something like this:

```js
if ( ! props.term ) {
	const validSearchTerm = props.searchTerm && props.searchTerm.trim().length;
	const callback = validSearchTerm ? this.isValid : null;

	this.setState( assign( {}, this.constructor.initialState, {
		name: validSearchTerm ? props.searchTerm : '',
	} ), callback );

	return;
}
```

It just gets a little convoluted because we only want to run `this.isValid` if we're actively setting the taxonomy name. Otherwise, clicking "Add New Tag" by default (with no search term) sets an empty name and tries to validate it like this:

<img width="553" alt="screen shot 2017-04-19 at 12 41 09 pm" src="https://cloud.githubusercontent.com/assets/7240478/25196305/842ab36a-24fd-11e7-8a04-150c5443a24f.png">

I thought breaking it out would be more clearer.